src/algokit/core/dispenser.py

```diff
@@ -0,0 +1,361 @@
+import base64
+import logging
+import os
+import time
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any
+
+import httpx
+import jwt
+import keyring
+from auth0.authentication.token_verifier import AsymmetricSignatureVerifier, TokenVerifier
+from cryptography.hazmat.primitives.asymmetric import rsa
+
+logger = logging.getLogger(__name__)
+
+# Constants
+ALGORITHMS = ["RS256"]
+DISPENSER_KEYRING_NAMESPACE = "algokit_dispenser"
+DISPENSER_KEYRING_ID_TOKEN_KEY = "algokit_dispenser_id_token"
+DISPENSER_KEYRING_ACCESS_TOKEN_KEY = "algokit_dispenser_access_token"
+DISPENSER_KEYRING_REFRESH_TOKEN_KEY = "algokit_dispenser_refresh_token"
+DISPENSER_KEYRING_USER_ID_KEY = "algokit_dispenser_user_id"
+DISPENSER_REQUEST_TIMEOUT = 15
+DISPENSER_ACCESS_TOKEN_KEY = "ALGOKIT_DISPENSER_ACCESS_TOKEN"
+
+
+class DispenserApiAudiences(str, Enum):
```

Comment:
In reading the docs I've realise that we have different terminology for these contexts.
User = interactive mode
CI = CI or non-interactive mode

Do we think there is any value in making the terminology consistent?

Reply:
Let's discuss after sync tomorrow @neilcampbell , i think it may be wrong terminology to refer to user as interactive mode and ci as non in the context of dispenser. Using any command (limit, fund, refund) with or without --ci token in our case is non interactive regardless, the ouput and behaviour for user and ci mode are identical its just that the authentication is implicit in case of ci. For other commands where you see interactive and etc that actually implies that somewhere within the flow there is a need for interaction with click cli where user has to provide some sort of response to a prompt. 

I agree though that we probably should make consistent terminology here so how about i remove mentions of interactive and non interactive in docs? Instead i can refer to `if such and such env var is provided, authentication will be attempted against that access token and etc`, simpler explanations on behaviour but, once again, no mentions of interactive/non-interactive cause its all non interactive in our case anyway

Reply:
@aorumbayev This makes sense. Let's update the docs.